fix(modal): add handling for missing reference element

max-umain opened the pull request:
## **Describe pull-request**  
Fixes the bug with missing reference element for modal.

## **Issue Linking:**  
- **GitHub:** https://github.com/scania-digital-design-system/tegel/issues/1164

## **How to test**  
1. Go to https://pr-233.d2vh2lmnzgzrkl.amplifyapp.com/web-components
2. Open the modal by hovering the icon below the text "Modals can also be opened programatically, see it in action by hovering the Icon below."
3. Click the exit button in the top right corner of the modal
4. Verify that it closes as expected
5. Go to the current react demo that has this bug, at https://react-demo.tegel.scania.com/web-components
6. Repeat steps 2-3
7. Note that it doesn't close

## **Checklist before submission**
- [ ] No accessibility violations in Storybook
- [ ] I have added unit tests for my changes (if applicable)
- [x] All existing tests pass
- [ ] I have updated the documentation (if applicable)
- [ ] Not breaking production behavior
- [ ] Behavior available in Storybook with documented descriptions (if applicable)
- [x] `npm run build:all` without errors

## **Suggested test steps**
- [ ] Browser testing (Chrome, Safari, Firefox) 
- [ ] Keyboard operability
- [ ] Interactive elements have labels.
- [ ] Storybook controls
- [ ] Design/controls/props is aligned with other components 
- [ ] Dark/light mode and variants 
- [ ] Input fields – values should be displayed properly 
- [ ] Events

## **Screenshots**  
None

## **Additional context**  
None

timrombergjakobsson commented on packages/core/src/components/modal/modal.tsx:
```diff
@@ -139,14 +150,21 @@ export class TdsModal {
     // If referenced element is a custom element eg: tds-button we find the interactive element inside
-    if(potentialReferenceElements.indexOf(referenceEl.tagName) < 0) {
-      referenceEl = referenceEl.querySelectorAll<HTMLElement>(potentialReferenceElements.join(','))[0]
+    if (potentialReferenceElements.indexOf(referenceElement.tagName) < 0) {
```
Maybe we can do:

```
const isNativeFocusable = potentialReferenceElements.includes(referenceElement.tagName);

if (!isNativeFocusable) {
  const interactiveEl = referenceElement.querySelector(potentialReferenceElements.join(','));
  if (interactiveEl) {
    interactiveEl.focus();
  }
} else {
  referenceElement.focus();
}

```
?

So we don't risk in calling `.focus()` on an undefined element.

theJohnnyMe replied:
Great suggestion @timrombergjakobsson !

timrombergjakobsson commented on packages/core/src/components/modal/modal.tsx:
```diff
@@ -106,6 +106,12 @@ export class TdsModal {
+
+    if (!this.selector && !this.referenceEl) {
+      console.warn(
+        'Tegel Modal component: please provide a selector or referenceEl prop to to reference the element used to show the modal',
```
```suggestion
        'Tegel Modal: Missing focus origin. Please provide either a "referenceEl" or a "selector" to ensure focus returns to the element that opened the modal.',
```

theJohnnyMe replied:
Maybe add part: Disregard this message if opening modal programatically.

max-umain commented:
Thank you @timrombergjakobsson for the feedback :) 

Fixed the code and updated the tegel-react-demo preview with a new beta release with the latest changes.

theJohnnyMe reviewed (approved):
LGTM
